src/components/Switch/Switch.tsx

```diff
@@ -1,44 +1,19 @@
-import React, { useCallback } from 'react'
-import styles from './styles.module.scss'
-
-export interface SwitchProps {
-	/**
-	 * Varient type of a switch
-	 * @default filled
-	 */
-	varient?: 'filled' | 'outlined'
-
-	/**
-	 * State of the switch
-	 * @default false
-	 */
-	value?: boolean
-
-	/**
-	 * Style applied.
-	 */
-	style?: React.CSSProperties
-
-	/**
-	 * Function to be called when switch is changed
-	 */
-	onChange: (value: boolean) => void
-}
-
-const Switch: React.FC<SwitchProps> = (props) => {
-	const onClick = useCallback(() => {
-		props.onChange(!props.value)
-	}, [props.onChange, props.value])
+import React from 'react';
+import './Switch.scss';
 
+const Switch:React.FC<React.InputHTMLAttributes<HTMLInputElement>> = (props) => {
 	return (
-		<label
-			className={`${styles.switch} ${props.value ? styles.active : ''} ${props.varient === 'outlined' ? styles.outlined : ''}`}
-			onClick={onClick}
-			style={props.style}
-		>
-			<span className={`${styles.round} ${styles.slider}`}></span>
+		<label className={`relative inline-block w-11 h-7 && ${props.disabled && 'bg-blue-300 opacity-20 rounded-full'}`}>
+			<input 
+                {...props}
+                disabled={props.disabled} 
```

Comment:
@manishdoley23 they are not needed as the props are already provided in the line above